### PR TITLE
[RESTEASY-2044] provided unit-tests for ExceptionHandler

### PR DESCRIPTION
**Pull request by rsearls**

### CODE REVIEW

File: testsuite/unit-tests/src/test/java/org/jboss/resteasy/test/mapper/ExceptionHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Test that the Exception processing and ExceptionMapper handling rules defined in
+ * the jar-311 section 3.3.4 are dealt with correctly.  Related information is provided
+ * in the Resteasy guide section 3.3.4
```
Comment (asoldano) [inline testsuite/unit-tests/src/test/java/org/jboss/resteasy/test/mapper/ExceptionHandlerTest.java]: I guess you're referring to section 30 of the resteasy doc, aren't you? http://docs.jboss.org/resteasy/docs/3.6.1.Final/userguide/html_single/index.html#ExceptionHandling

Comment (asoldano) [reply testsuite/unit-tests/src/test/java/org/jboss/resteasy/test/mapper/ExceptionHandlerTest.java]: and please s/jar-311/JSR-311/

Comment (asoldano) [inline testsuite/unit-tests/src/test/java/org/jboss/resteasy/test/mapper/ExceptionHandlerTest.java]: I assume you're referring to section 30 of the doc, http://docs.jboss.org/resteasy/docs/3.6.1.Final/userguide/html_single/index.html#ExceptionHandling , right?
And please turn "jar-311" to "JSR-311"

Comment (rsearls) [reply testsuite/unit-tests/src/test/java/org/jboss/resteasy/test/mapper/ExceptionHandlerTest.java]: Its seciton 29 in mater's docbook, resteasy-reference-guide-en-US.pdf

Fixed

Comment (rsearls) [reply testsuite/unit-tests/src/test/java/org/jboss/resteasy/test/mapper/ExceptionHandlerTest.java]: Its seciton 29 in mater's docbook, resteasy-reference-guide-en-US.pdf

Fixed

**Pull request merged**